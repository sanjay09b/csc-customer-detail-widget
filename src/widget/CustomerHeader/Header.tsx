import React, { useEffect, useState } from "react";
import axios from "axios";
import VerticalLine from "../SeparationLine/VerticalLine";
import RedirectButton from "../RedirectButton/RedirectButton";
import { CustomerHeader } from "../Interface";
// import StatusButton from "../Utility/StatusButton";
import ActiveStatusButton from "../StatusButtons/ActiveStatusButton";
import DeactivatedStatusButton from "../StatusButtons/DeactivatedStatusButton";
import FraudStatusButton from "../StatusButtons/FraudStatusButton";
import BannedStatusButton from "../StatusButtons/BannedStatusButton";
import SuspendedStatusButton from "../StatusButtons/SuspendedStatusButton";
import CustomStyle from "../CustomStyles";

const statusButton = (status): JSX.Element | null => {
  switch (status) {
    case "Active":
      return <ActiveStatusButton />;
    case "Deactivated":
      return <DeactivatedStatusButton />;
    case "Fraud":
      return <FraudStatusButton />;
    case "Banned":
      return <BannedStatusButton />;
    case "Suspended":
      return <SuspendedStatusButton />;
    default:
      return <div>Unknown Error</div>;
  }
};

const Header = (): JSX.Element => {
  const [userData, setUserData] = useState<CustomerHeader>();
  const [loading, setLoading] = useState(true);

  useEffect((): void => {
    const url: string =
      "https://6409df426ecd4f9e18bdd647.mockapi.io/api/v1/customers/641016776/customer";
    async function getUserData() {
      try {
        if (loading) {
          const response = await axios.get(url);
          setUserData(response.data.customerDetails[0]);
        }
      } catch (error) {
        console.log(error);
      } finally {
        setLoading(false);
      }
    }
    getUserData();
  }, []);

  return (
    <>
      {loading ? (
        <div>Loading...</div>
      ) : (
        <>
          <div className={CustomStyle.customerHeader.alignRow}>
            <div className={CustomStyle.customerHeader.nameStatusWrapper}>
              <div className={CustomStyle.customerHeader.name}>
                {userData.firstName} {userData.lastName}
              </div>
              {statusButton(userData.activeStatus)}
            </div>
            <div className={CustomStyle.customerHeader.redirectPositionWrapper}>
              <div
                className={CustomStyle.customerHeader.sephoraRedirectWrapper}
              >
                <span className={CustomStyle.customerHeader.sephoraRedirect}>
                  Visit client on Sephora.com
                </span>
                <RedirectButton />
              </div>
            </div>
          </div>
          <div className={CustomStyle.customerHeader.alignRowWithSpace}>
            <div className={CustomStyle.customerHeader.centerAlign}>
              {userData.email}
            </div>
            <VerticalLine />
            <div className={CustomStyle.customerHeader.centerAlign}>
              {userData.phoneNumber}
            </div>
            <VerticalLine />
            <div className={CustomStyle.customerHeader.centerAlign}>
              {userData.risk}
            </div>
          </div>
        </>
      )}
    </>
  );
};

export { Header, statusButton };
