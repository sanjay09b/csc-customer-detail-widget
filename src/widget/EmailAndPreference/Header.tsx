import React from "react";

const Header = () => {
  return (
    <>
      <div className="font-bold text-xl mx-4 my-3">
        Email & Mail Preferences
      </div>
    </>
  );
};

export default Header;
