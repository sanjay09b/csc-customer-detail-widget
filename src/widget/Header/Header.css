.header-wrapper {

    font-size: 28px;
}