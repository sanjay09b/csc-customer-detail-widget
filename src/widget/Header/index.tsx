import React from "react";
import "./Header.css";

const CustomerHeader = () => {
  return <div className="header-wrapper"> CustomerHeader Component</div>;
};

export default CustomerHeader;
