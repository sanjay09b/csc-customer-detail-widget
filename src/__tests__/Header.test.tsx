import React from "react";
import { render, waitFor, screen } from "@testing-library/react";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import Header from "../widget/CustomerHeader/Header";

const mockUserData = {
  firstName: "John",
  lastName: "Doe",
  email: "john.doe@example.com",
  phoneNumber: "+1 (555) 123-4567",
  risk: "Low",
  activeStatus: "Active",
};

const mockApiUrl =
  "https://6409df426ecd4f9e18bdd647.mockapi.io/api/v1/customers/641016776/customer";

describe("Header component", () => {
  let mockAxios: MockAdapter;

  beforeEach(() => {
    mockAxios = new MockAdapter(axios);
  });

  afterEach(() => {
    mockAxios.reset();
  });

  it("renders the user data when the API request is successful", async () => {
    mockAxios.onGet(mockApiUrl).reply(200, { customerDetails: [mockUserData] });

    render(<Header />);

    await waitFor(() => expect(mockAxios.history.get.length).toBe(1));

    expect(screen.queryByText("Loading...")).not.toBeInTheDocument();
    expect(
      screen.getByText(`${mockUserData.firstName} ${mockUserData.lastName}`),
    ).toBeInTheDocument();
    expect(screen.getByText(`Visit client on Sephora.com`)).toBeInTheDocument();
    expect(screen.getByText(`${mockUserData.email}`)).toBeInTheDocument();
    expect(screen.getByText(`${mockUserData.phoneNumber}`)).toBeInTheDocument();
    expect(screen.getByText(`${mockUserData.risk}`)).toBeInTheDocument();
  });
});
