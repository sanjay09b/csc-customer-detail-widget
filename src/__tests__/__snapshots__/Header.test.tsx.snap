// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the correct status button based on the user's active status 1`] = `<ActiveStatusButton />`;

exports[`Header component renders the correct status button based on the user's active status 2`] = `<DeactivatedStatusButton />`;

exports[`Header component renders the correct status button based on the user's active status 3`] = `<FraudStatusButton />`;

exports[`Header component renders the correct status button based on the user's active status 4`] = `<BannedStatusButton />`;

exports[`Header component renders the correct status button based on the user's active status 5`] = `<SuspendedStatusButton />`;

exports[`Header component renders the correct status button based on the user's active status 6`] = `
<div>
  Unknown Error
</div>
`;
